charts/internal/cloud-provider-config/templates/_cloud-provider-config-route.tpl

```diff
@@ -0,0 +1,6 @@
+{{- define "cloud-provider-config-route" -}}
+[Route]
+{{- if .Values.routerID }}
+router-id="{{ .Values.routerID }}"
```

Comment:
There is also one more failure mode which is particular to openstack: 

Because now we do configure the cloud routes, there are rare cases where the CCM is terminated before getting to clear all the routes during shoot termination. Because there are still custom routes in the router, the deletion of the router cannot proceed.

You can see that for other cloud providers e.g. gcp ([ref](https://github.com/gardener/gardener-extension-provider-gcp/blob/9992d3829cdea7071c9c50318c17debad3969677/pkg/controller/infrastructure/actuator_delete.go#L132)) we do have custom cleaning logic that makes sure that this doesn't happen.